Expensify/App
# Add attachment picker to Expensify Chat

@tgolen @AndrewGable will you please review this?

Fixes https://github.com/Expensify/ReactNativeChat/issues/87

### Details
This PR adds support for uploading images on iOS/Android, and files on web.
[This issue](https://github.com/facebook/react-native/issues/29364) still exists on React Native version 0.63+, so to fix this I've added the [patch-package](https://www.npmjs.com/package/patch-package) dependency to allow us to apply the 1 line fix for this issue rather than dealing with reverting the RN version to 0.62.2, which requires that we manually manage pods in the podfile instead of using `use_react_native!(:path => config["reactNativePath"])` to manage all of them. Once [the fix for the RN issue](https://github.com/facebook/react-native/pull/29595) is merged, we can simply remove the `patch-package` line in package.json, and remove the `patch` directory.

### Tests
1. Go to a report on web and hit the attachment icon, confirm a file picker shows.
2. Upload a file, confirm you see the attachment loading placeholder.
3. Confirm after the file uploads, the thumbnail image displays properly
4. Hit the attachment icon on iOS/Android, confirm you can upload an image from the photo library, and the attachment loading placeholder shows while the file is uploading
5. Hit the attachment icon on iOS/Android, confirm you can upload an image by taking a picture, and the attachment loading placeholder shows while the file is uploading
